### PR TITLE
pre-commit command ran for humitos/log/sentry branch

### PR DESCRIPTION
**Pull request by humitos**
Logic changes are under #3893

### CODE REVIEW

File: readthedocs/builds/syncers.py [hunk 1]
```diff
@@ -53,30 +54,26 @@ def copy(cls, path, target, is_file=False, **__):
...
+                    slash = '/'
                 # Add a slash when copying directories
                 sync_cmd = (
                     "rsync -e 'ssh -T' -av --delete {path}{slash} {user}@{server}:{target}"
                     .format(
-                        path=path,
-                        slash=slash,
-                        user=sync_user,
-                        server=server,
+                        path=path, slash=slash, user=sync_user, server=server,
```
Comment (agjohnson) [inline readthedocs/builds/syncers.py]: This squashed the continuation

File: readthedocs/builds/syncers.py [hunk 2]
```diff
@@ -91,31 +88,25 @@ def copy(cls, path, target, host, is_file=False, **__):
...
             # Add a slash when copying directories
             sync_cmd = (
                 "ssh {user}@{server} 'rsync -av "
                 "--delete --exclude projects {user}@{host}:{path} {target}'"
                 .format(
-                    host=host,
-                    path=path,
-                    user=sync_user,
-                    server=server,
+                    host=host, path=path, user=sync_user, server=server,
```
Comment (agjohnson) [inline readthedocs/builds/syncers.py]: Squashed continuation here

File: readthedocs/core/management/commands/clean_builds.py [hunk 3]
```diff
@@ -18,40 +21,41 @@ class Command(BaseCommand):
...
         max_date = datetime.now() - timedelta(days=options['days'])
-        queryset = (Build.objects
-                    .values('project', 'version')
-                    .annotate(max_date=Max('date'))
-                    .filter(max_date__lt=max_date)
-                    .order_by('-max_date'))
+        queryset = (
+            Build.objects.values('project',
+                                 'version').annotate(max_date=Max('date'))
+            .filter(max_date__lt=max_date).order_by('-max_date'))
```
Comment (agjohnson) [inline readthedocs/core/management/commands/clean_builds.py]: Previous continuation style was better

File: readthedocs/core/views/hooks.py [hunk 4]
```diff
@@ -173,12 +170,14 @@ def github_build(request):  # noqa: D205
...
-            if request.META['CONTENT_TYPE'] == 'application/x-www-form-urlencoded':
+            if request.META[
+                    'CONTENT_TYPE'] == 'application/x-www-form-urlencoded':
                 data = json.loads(request.POST.get('payload'))
             else:
                 data = json.loads(request.body)
             http_url = data['repository']['url']
-            http_search_url = http_url.replace('http://', '').replace('https://', '')
+            http_search_url = http_url.replace('http://',
+                                               '').replace('https://', '')
```
Comment (agjohnson) [inline readthedocs/core/views/hooks.py]: I'd prefer a different method of breaking this up to multiple lines

File: readthedocs/builds/syncers.py (hunk 1 above)
Comment (humitos) [reply readthedocs/builds/syncers.py]: I had sent a comment here, but my internet connection didn't help me:

----

By adding a `,` after the `target` here, we get a one attribute per line formatting automatically by yapf.

There are many changes on these files were we can apply this fix to get the more readable way, but I don't want to do it manually for now.

I think we can easily add the trailing comman once we start using pre-commit in a daily basis.

----

Anyway, I think we can use this PR to discuss/tweak yapf a little more.

There are things that need manual work. Although, for this particular issue that the trailling comma is missing we could add automatically before calling yapf with this hook: https://github.com/asottile/add-trailing-comma

**Comment by humitos** [comment]
I re-configure `pre-commit` (depends on this branch https://github.com/rtfd/common/pull/2) and re-run it to show what are the news here.

I found one problem though, the definition of the functions is also affected by `add-trailing-comma` plugin, so I will see if we can disable that.

Anyway, it seems to be too much closer to what you want.

File: readthedocs/builds/models.py [hunk 5]
```diff
@@ -247,13 +273,18 @@ def clean_build_path(self):
...
         except OSError:
             log.exception('Build path cleanup failed')
 
     def get_github_url(
-            self, docroot, filename, source_suffix='.rst', action='view'):
+            self,
+            docroot,
+            filename,
+            source_suffix='.rst',
+            action='view',
```
Comment (humitos) [inline readthedocs/builds/models.py]: Here, the function is affected by `add-trailing-comma` plugin.

I want to check how to disable this case.

File: readthedocs/core/views/hooks.py [hunk 6]
```diff
@@ -126,25 +136,24 @@ def _build_url(url, projects, branches):
             # Call SyncRepositoryTask to update tag/branch info
             version = project.versions.get(slug=LATEST)
             sync_repository = SyncRepositoryTask()
-            sync_repository.apply_async(
-                args=(version.pk,),
-            )
+            sync_repository.apply_async(args=(version.pk,),)
```
Comment (humitos) [inline readthedocs/core/views/hooks.py]: It seems it gets confused here. Not a big issue, though.

File: readthedocs/core/views/hooks.py [hunk 7]
```diff
@@ -173,12 +182,16 @@ def github_build(request):  # noqa: D205
     """
     if request.method == 'POST':
         try:
-            if request.META['CONTENT_TYPE'] == 'application/x-www-form-urlencoded':
+            if request.META['CONTENT_TYPE'
+                            ] == 'application/x-www-form-urlencoded':
```
Comment (humitos) [inline readthedocs/core/views/hooks.py]: This is one is terrible.

Comment (humitos) [reply readthedocs/core/views/hooks.py]: We could add `# yapf: disable` here, since it's a special case.

File: readthedocs/core/views/hooks.py [hunk 8]
```diff
@@ -267,25 +281,28 @@ def bitbucket_build(request):
...
                 data = json.loads(request.POST.get('payload'))
             else:
                 data = json.loads(request.body)
 
-            version = 2 if request.META.get('HTTP_USER_AGENT') == 'Bitbucket-Webhooks/2.0' else 1
+            version = 2 if request.META.get(
+                'HTTP_USER_AGENT',
+            ) == 'Bitbucket-Webhooks/2.0' else 1
```
Comment (humitos) [inline readthedocs/core/views/hooks.py]: This one is also terrible.

Comment (humitos) [reply readthedocs/core/views/hooks.py]: There are some `_PENALTY` settings that we should take a look for this case.

File: readthedocs/payments/forms.py [hunk 9]
```diff
@@ -182,13 +207,12 @@ def clear_card_data(self):
...
+            raise AttributeError(
+                'Form was passed immutable QueryDict POST data',
+            )
 
     def fields_with_cc_group(self):
-        group = {
-            'is_cc_group': True,
-            'fields': []
-        }
+        group = {'is_cc_group': True, 'fields': []}
```
Comment (humitos) [inline readthedocs/payments/forms.py]: Here it should have added the `,` and formatted it as it was.

File: readthedocs/search/parse_json.py [hunk 10]
```diff
@@ -16,16 +17,22 @@
...
+            include_file=False,
+        )
     html_files = []
     for root, _, files in os.walk(full_path):
         for filename in fnmatch.filter(files, '*.fjson'):
-            if filename in ['search.fjson', 'genindex.fjson', 'py-modindex.fjson']:
+            if filename in [
+                    'search.fjson',
+                    'genindex.fjson',
+                    'py-modindex.fjson', ]:
```
Comment (humitos) [inline readthedocs/search/parse_json.py]: Also weird here

File: readthedocs/builds/models.py (hunk 5 above)
Comment (humitos) [reply readthedocs/builds/models.py]: This is not a problem of this plugin, but a combination of `add-trailing-comma` plus `yapf`.

When this is detected

```
 def myfunc(
        a, b, c, d, e, f, argument, larger, here, to, reach, eighty, lines, NOW
 ):
```

trailing comma adds a comma in the end, and yapf format it on an argument per line in the next pass.

**Comment by humitos** [comment]
> I found one problem though, the definition of the functions is also affected by add-trailing-comma plugin, so I will see if we can disable that.

This is not possible to disable in this plugin. I opened an issue in yapf to see why it's not respecting a setting: https://github.com/google/yapf/issues/551

**Comment by ericholscher** [comment]
Pretty sure we can close this, as it has a million conflicts.

**Comment by ericholscher** [comment]
Actually not sure -- what should we do here?

**Comment by humitos** [comment]
Let's close it.

We still don't have a strict policy regarding how to ues `pre-commit`. We are not forcing nobody to run this command and because nobody is using it in a daily basis it may need some more work now.

We can come back to this later with more time.

**Pull request closed without merging**